src/MSALWrapper/AuthFlow/DeviceCode.cs

```diff
@@ -63,56 +64,45 @@ public DeviceCode(ILogger logger, Guid clientId, Guid tenantId, IEnumerable<stri
         /// <inheritdoc/>
         protected override async Task<(TokenResult, IList<Exception>)> GetTokenInnerAsync()
         {
-            IAccount account = await this.pcaWrapper.TryToGetCachedAccountAsync(this.preferredDomain) ?? null;
-
-            if (account != null)
-            {
-                this.logger.LogDebug($"Using cached account '{account.Username}'");
-            }
+            IAccount account = await this.pcaWrapper.TryToGetCachedAccountAsync(this.preferredDomain);
+            TokenResult tokenResult = null;
 
             try
             {
-                try
+                tokenResult = await CachedAuth.GetTokenAsync(
+                    this.logger,
+                    this.scopes,
+                    account,
+                    this.pcaWrapper,
+                    this.errors);
+
+                if (tokenResult != null)
                 {
-                    var tokenResult = await TaskExecutor.CompleteWithin(
-                        this.logger,
-                        this.silentAuthTimeout,
-                        "Get Token Silent",
-                        (cancellationToken) => this.pcaWrapper.GetTokenSilentAsync(
-                            this.scopes,
-                            account,
-                            cancellationToken),
-                        this.errors)
-                        .ConfigureAwait(false);
-                    tokenResult.SetSilent();
-
                     return (tokenResult, this.errors);
                 }
-                catch (MsalUiRequiredException ex)
-                {
-                    this.errors.Add(ex);
-                    this.logger.LogDebug($"Silent auth failed, re-auth is required.\n{ex.Message}");
-                    var tokenResult = await TaskExecutor.CompleteWithin(
-                        this.logger,
-                        this.deviceCodeFlowTimeout,
-                        "Get Token using Device Code",
-                        (cancellationToken) => this.pcaWrapper.GetTokenDeviceCodeAsync(
+
+                this.logger.LogWarning($"Device Code Authentication for: {this.promptHint}");

```

Comment:
Same comment as previous PRs. We should assert these log statements in tests.

Reply:
Resolving with #282